clar2wasm/tests/wasm-generation/principal.rs

```diff
@@ -0,0 +1,231 @@
+                Value::Sequence(SequenceData::String(CharType::ASCII(ASCIIData {
+                    data: ctc.into_bytes()
+                })))
+            ));
+
+            let expected = match version_byte {
+                // Valid range for version_bytes
+                0x00..=0x1f => {
+                    match version_byte {
+                        // Since tests runs on a Testnet version,
+                        // version_bytes single_sig (0x1A) || multi_sig (0x15), for Testnet,
+                        // will return an Ok value.
+                        0x1A | 0x15 => Value::okay(
+                            TupleData::from_data(vec![
+                                ("hash-bytes".into(), hash_bytes),
+                                ("name".into(), Value::Optional(OptionalData { data })),
+                                (
+                                    "version".into(),
+                                    Value::Sequence(SequenceData::Buffer(BuffData {
+                                        data: vec![version_byte as u8],
+                                    })),
+                                ),
+                            ])
+                            .unwrap()
+                            .into()
+                        ),
+                        _ => Ok(create_error_destruct(hash_bytes, version_byte as u8, data)),
+                    }
+                },
+                _ => Ok(create_error_destruct(hash_bytes, version_byte as u8, data)),
+            }.unwrap();
+
+            crosscheck(
+                &format!("(principal-destruct? {})", PropValue::from(expected_principal.clone())),
+                Ok(Some(expected)),
+            );
+        }
+    }
+
+    proptest! {
+        #![proptest_config(runtime_config())]
+
+        #[test]
+        fn crosscheck_is_standard(
+            version_byte in 0x00..=0x1f,
+            hash_bytes in buffer(20),
+            contract in "([a-zA-Z](([a-zA-Z0-9]|[-])){0, 30})".prop_flat_map(|name| {
+                prop_oneof![Just(Some(name)), Just(None)]
+            })
+        ) {
+            let expected_principal = create_principal(
+                version_byte as u8,
+                &hash_bytes.expect_buff(20).unwrap(),
+                contract.as_deref()
+            );
+
+            let principal_str = PropValue::from(expected_principal.clone()).to_string();
+            let expected = matches!(principal_str.get(0..3), Some("'ST") | Some("'SN"));
+
+            crosscheck(
+                &format!("(is-standard {})", PropValue::from(expected_principal.clone())),
+                Ok(Some(Value::Bool(expected))),
+            );
+        }
+    }
+
+    proptest! {
+        #![proptest_config(runtime_config())]
+
+        #[test]
+        fn crosscheck_principal_of(
+            key in buffer(33)
+        ) {
+            let hex_string = &format!("{}", key)[2..];
+            let pubkey = Secp256k1PublicKey::from_hex(hex_string);
```

Comment:
How about retrieving the bytes from the Value directly and using `from_slice` instead?

Reply:
Implementation has changed.